### PR TITLE
End of Game Logic Finally Fixed

### LINKED ISSUE #198: Make the round manager work properly
**Describe the bug**
Just make it work pls.

### PR DESCRIPTION
**Pull request by ChrisGora**
Fixed the game logic so that the Game always ends when we expect it to. Turns out Round Manager was correct, Parking Manager a bit less so.

The Game correctly ends but the disconnect does not happen.

Smaller changes:
- Client and Server Parking Managers are their own files.
-  The HUD no longer shows any placeholders (and appears to work correctly all the time).

Resolves #198 
Made some progress on #199
#197 is not done

### CODE REVIEW

**Comment by RileyEv** [comment]
Okay so for this is it just that the `Server` isn't handling the game end event correctly?

**Comment by RileyEv** [comment]
Does the eliminating player stuff work now too? or is that just a disconnect problem too 🙈

**Comment by ChrisGora** [comment]
Just tested this. So, in a round with 3 players:

- 2 Players claimed a space.
- The player without a space gets despawned but not disconnected.
- An exception happened, see #226
- Round Ended
- The other 2 players continued in a new round
- Neither of them claimed a space
- They both got told they lost (but they don't appear despawned nor disconnected)
- (and the server doesn't start another round which it did before)

**TLDR It all works**

**Comment by RileyEv** [comment]
There is a chance, I changed it so that players aren't disconnected anymore 🤔 because they'll turn to police cars. But need to check that one in a minute

**Review by RileyEv — APPROVED** [review]
